cheroot/workers/threadpool.py

```diff
@@ -173,17 +173,7 @@ def __init__(
 
     def start(self):
         """Start the pool of threads."""
-        for _ in range(self.min):
-            self._threads.append(WorkerThread(self.server))
-        for worker in self._threads:
-            worker.name = (
-                'CP Server {worker_name!s}'.
-                format(worker_name=worker.name)
-            )
-            worker.start()
-        for worker in self._threads:
-            while not worker.ready:
-                time.sleep(.1)
```

Comment:
The waiting implementation here, in the original code, seems a bit more efficient, while `grow()` has more looping over the data as it seems.

Have you compared the performance of both?

Reply:
I have not compared the performance between using the original code from `start()` and the code from `grow()` but have some thoughts:
 - Both the current implementation and `grow()` wait for all the threads to be started and ready with a spinlock delay of 100ms. Compared with variance introduced by that, I would imagine that any difference in performance of looping over a datastructure of this size a few times to be negligible. With that being said, I've changed `grow()` to use the structure from `start()` since I agree with you that it should be faster.
 - This code is run in the `__init__` of a `ThreadPool` which is only initialized twice in the entire codebase, both times in the `__init__` of long-lived objects (HTTP servers). It's not on the critical path or I would've paid more attention to micro-optimizing it.

Reply:
Okay, that sounds reasonable. Marking this thread as resolved.